### PR TITLE
[Feature] Split line items into new, modified, and deleted

### PR DESCRIPTION
**Pull request by mikkoh**
This PR creates 3 lists of line items on `CartLineItems`:

* New line items
* Modified line items
* Deleted line items

This will allow us build queries to create, modify, delete line items in the Storefront API.

### CODE REVIEW

File: Assets/Editor/TestCart.cs [hunk 1]
```diff
@@ -272,5 +272,81 @@ public class TestCart {
...
+            cart.LineItems.Delete(variantId3);
+
+            Assert.AreEqual(0, cart.LineItems.New().Count, "No new line items");
+            Assert.AreEqual(0, cart.LineItems.Modified().Count, "No modified line items");
+            Assert.AreEqual(1, cart.LineItems.Deleted().Count, "Has one deleted line item");
+
+            cart.LineItems.ClearDeleted();
+
+            Assert.AreEqual(0, cart.LineItems.Deleted().Count, "Has 0 deleted line item after clear");
+            Assert.AreEqual(1, cart.LineItems.All().Count, "Has 1 line item at end");
```
Comment (dengjeffrey) [inline Assets/Editor/TestCart.cs]: Should we also be testing if the remaining items are what is suppose to be in the cart?

File: scripts/generator/graphql_generator/csharp/SDK/CartLineItems.cs.erb [hunk 2]
```diff
@@ -127,28 +130,38 @@ namespace <%= namespace %>.SDK {
...
                 if (quantity != null) {
-                    input.Quantity = (long) quantity;
+                    modifiedLineItem.Quantity = (long) quantity;
+                    didModify = true;
                 }
 
                 if (customAttributes != null) {
-                    input.CustomAttributes = customAttributes;
+                    modifiedLineItem.CustomAttributes = customAttributes;
+                    didModify = false;
```
Comment (dengjeffrey) [inline scripts/generator/graphql_generator/csharp/SDK/CartLineItems.cs.erb]: Could you quickly explain what custom attributes are?  Are line items not considered modified if we update custom attributes?

Comment (mikkoh) [inline scripts/generator/graphql_generator/csharp/SDK/CartLineItems.cs.erb]: Sorry do you mean directly modify the `CustomAttributes` or overwrite like this case?

File: Assets/Editor/TestCart.cs (hunk 1 above)
Comment (mikkoh) [inline Assets/Editor/TestCart.cs]: This may already be tested in another test but I'll double check.

Comment (mikkoh) [inline Assets/Editor/TestCart.cs]: I ended up adding in this test. Better safe than sorry.

**Comment by mikkoh** [comment]
@dengjeffrey I've fixed the above issues. I ended up making watching line item changes more robust where the line item itself notifies `CartLineItems` that it has been changed.

**Pull request merged**